runtime/gc_modron_standard/StandardAccessBarrier.cpp

```diff
@@ -92,13 +104,122 @@ MM_StandardAccessBarrier::tearDown(MM_EnvironmentBase *env)
+			oldObject = mmPointerFromToken(vmThread, *destAddress);
+			protectIfVolatileAfter(vmThread, isVolatile, true, false);
+			rememberObjectToRescan(env, oldObject);
+		}
+	}
+
+	return true;
+}
+
+bool
+MM_StandardAccessBarrier::preObjectStoreInternal(J9VMThread *vmThread, J9Object **destAddress, J9Object *value, bool isVolatile)
+{
+	MM_EnvironmentBase* env = MM_EnvironmentBase::getEnvironment(vmThread->omrVMThread);
+
+	if (isSATBBarrierActive(env)) {
+		if (isDoubleBarrierActiveOnThread(vmThread)) {
+			rememberObjectToRescan(env, value);
+		}
+		J9Object* oldObject = NULL;
+		protectIfVolatileBefore(vmThread, isVolatile, true, false);
+		oldObject = *destAddress;
+		protectIfVolatileAfter(vmThread, isVolatile, true, false);
+		rememberObjectToRescan(env, oldObject);
+	}
+
+	return true;
+}
+/**
+ * @copydoc MM_ObjectAccessBarrier::preObjectStore()
+ *
+ * Metronome uses a snapshot-at-the-beginning algorithm, but with a fuzzy snapshot in the
+ * sense that threads are allowed to run during the root scan.  This requires a "double
+ * barrier."  The barrier is active from the start of root scanning through the end of
+ * tracing.  For an unscanned thread performing a store, the new value is remembered by
+ * the collector.  For any thread performing a store (whether scanned or not), the old
+ * value is remembered by the collector before being overwritten (thus this barrier must be
+ * positioned as a pre-store barrier).  For the latter ("Yuasa barrier") aspect of the
+ * double barrier, only the first overwritten value needs to be remembered (remembering
+ * others is harmless but not needed), and so we omit synchronization on the reading of the
+ * old value.
+ **/
+bool
+MM_StandardAccessBarrier::preObjectStore(J9VMThread *vmThread, J9Object *destObject, fj9object_t *destAddress, J9Object *value, bool isVolatile)
+{
+	return preObjectStoreInternal(vmThread, destObject, destAddress, value, isVolatile);
+}
+
+/**
+ * @copydoc MM_MetronomeAccessBarrier::preObjectStore()
+ *
+ * Used for stores into classes
+ */
+bool
+MM_StandardAccessBarrier::preObjectStore(J9VMThread *vmThread, J9Object *destClass, J9Object **destAddress, J9Object *value, bool isVolatile)
+{
+	return preObjectStoreInternal(vmThread, destAddress, value, isVolatile);
+}
+
+/**
+ * @copydoc MM_MetronomeAccessBarrier::preObjectStore()
+ *
+ * Used for stores into internal structures
+ */
+bool
+MM_StandardAccessBarrier::preObjectStore(J9VMThread *vmThread, J9Object **destAddress, J9Object *value, bool isVolatile)
+{
+	return preObjectStoreInternal(vmThread, destAddress, value, isVolatile);
+}
 /**
  * Called after an object is stored into another object.
  */
 void
 MM_StandardAccessBarrier::postObjectStore(J9VMThread *vmThread, J9Object *destObject, fj9object_t *destAddress, J9Object *value, bool isVolatile)
 {
-	postObjectStoreImpl(vmThread, destObject, value);
+	if (_extensions->configuration->isIncrementalUpdateBarrierEnabled()) {
```

Comment:
similarly to isSATBBarrierActive, there should be a helper isIncrementalUpdateBarrierActive within this class.
then, that helper  probably does not need to go as far as to configuration, but just check in thread local data if the barrier is active. it should not ever be really active on thread level, if in configuration it's not enabled

Reply:
for now moved it to a helper with isIncrementalUpdateBarrierEnabled  check.